tests/workflows/pw/test_base.py

```diff
@@ -130,6 +130,34 @@ def test_handle_relax_recoverable_ionic_convergence_error(generate_workchain_pw,
     assert result.status == 0
 
 
+def test_handle_electronic_convergence_warning(generate_inputs_pw, generate_workchain_pw, generate_structure):
+    """Test `PwBaseWorkChain.handle_electronic_convergence_warning`."""
+    from aiida.common.links import LinkType
+
+    inputs = generate_inputs_pw()
```

Comment:
This is why the test fails. This fixture generates inputs for the `PwCalculation` not the `PwBaseWorkChain`. Use the following instead:
```
inputs = generate_workchain_pw(return_inputs=True)
```
that generates inputs for the workchain and returns the inputs instead of creating the instance, giving you the opportunity to update them as you like. Then call `generate_workchain_pw` again.

Reply:
Great, thanks!

The error does come from `aiida-pseudo`:
```
ERROR tests/workflows/pw/test_base.py::test_handle_electronic_convergence_warning - ValueError: md5 does not match that of stored file: 62f7fc0614e61f39d2530dbfe4ec53b6 != d41d8cd98f00b204e9800998ecf8427e
```

```
____________________________________________________________________________________________________________________________ ERROR at setup of test_handle_electronic_convergence_warning ____________________________________________________________________________________________________________________________

aiida_profile = <aiida.manage.tests.TestManager object at 0x7f0047c85640>, generate_upf_data = <function generate_upf_data.<locals>._generate_upf_data at 0x7f00429fab80>

    @pytest.fixture(scope='session', autouse=True)
    def sssp(aiida_profile, generate_upf_data):
        """Create an SSSP pseudo potential family from scratch."""
        from aiida.common.constants import elements
        from aiida.plugins import GroupFactory
    
        aiida_profile.reset_db()
    
        SsspFamily = GroupFactory('pseudo.family.sssp')
    
        cutoffs = {}
        stringency = 'standard'
    
        with tempfile.TemporaryDirectory() as dirpath:
            for values in elements.values():
    
                element = values['symbol']
>               upf = generate_upf_data(element)

tests/conftest.py:128: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/conftest.py:295: in _generate_upf_data
    return UpfData(file=handle)
../aiida-core/aiida/orm/nodes/data/singlefile.py:48: in __init__
    self.set_file(file)
../aiida-pseudo/aiida_pseudo/data/pseudo/upf.py:86: in set_file
    super().set_file(source, filename, **kwargs)
../aiida-pseudo/aiida_pseudo/data/pseudo/pseudo.py:134: in set_file
    self.md5 = md5_from_filelike(source)
../aiida-pseudo/aiida_pseudo/data/pseudo/pseudo.py:186: in md5
    self.validate_md5(value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <UpfData: uuid: 457be4f1-e194-417f-9727-9b210445ea43 (unstored)>, md5 = '62f7fc0614e61f39d2530dbfe4ec53b6'

    def validate_md5(self, md5: str):
        """Validate that the md5 checksum matches that of the currently stored file.
    
        :param value: the md5 checksum.
        :raises ValueError: if the md5 does not match that of the currently stored file.
        """
        with self.open(mode='rb') as handle:
            md5_file = md5_from_filelike(handle)
            if md5 != md5_file:
>               raise ValueError(f'md5 does not match that of stored file: {md5} != {md5_file}')
E               ValueError: md5 does not match that of stored file: 62f7fc0614e61f39d2530dbfe4ec53b6 != d41d8cd98f00b204e9800998ecf8427e

../aiida-pseudo/aiida_pseudo/data/pseudo/pseudo.py:111: ValueError
```

Reply:
Ah, this was a bug in our fixtures in `conftest.py`. I already fixed it, just rebase on latest `develop` and you should be good. Good lesson to always update your working branches ;)